### PR TITLE
Pipeline for generating isoscapes from a pre-compiled variational model

### PR DESCRIPTION
**Pull request by VimWizardVersace**

### CODE REVIEW

**Comment by VimWizardVersace** [comment]
> We want to move this to raster.py as it is a utility we want to have there and de-dupe (in those cases) if there are existing things to use (for ease of reference), as well as test the things you're coding here to make sure the code is resilient.

Done. I moved everything (except the model/scaler loading code) to raster.py.

I don't think there was any overlap with anything in raster.py. I'm actually really surprised get_prediction_at_each_pixel wasn't implemented anywhere yet.

File: raster.py [hunk 1]
```diff
@@ -263,6 +267,58 @@ def get_data_at_coords(dataset: AmazonGeoTiff, x: float, y: float, month: int) -
...
+      row["long"] = x
+
+      # Surround in try/except as we will be trying to fetch out of bounds data.
+      try:
+        for geotiff_label, geotiff in geotiffs.items():
+          row[geotiff_label] = geotiff.value_at(x, y)
+          if pd.isnull(row[geotiff_label]):
+            raise ValueError
+      except (ValueError, IndexError):
+        continue # masked and out-of-bounds coordinates
```
Comment (benwulfe) [inline raster.py:298]: you'll want to at least print out an error message indicating this has occurred.

Comment (VimWizardVersace) [reply raster.py:299]: Actually, this exception is used for coords outside the bounds of brazil unfortunately. The print statement would be really noisy.

Also, there shouldn't be an exception here.

**Comment by VimWizardVersace** [comment]
> Looks good. Just 2 requested changes.
> 
> Create a AmazonModel to wrap a keras.Model to run prediction and keep tf out as a dependency Create constant strings for the raster names that you have in your dict that returns functions. Ideally, discoverable in a class.

Both done.

File: raster.py [hunk 2]
```diff
@@ -308,5 +362,115 @@ def cellulose_isoscape_geotiff() -> AmazonGeoTiff:
...
+
+d13C_var_geotiff_ = None
+def d13C_var_geotiff() -> AmazonGeoTiff:
+  global d13C_var_geotiff_
+  if not d13C_var_geotiff_:
+    d13C_var_geotiff_ = load_named_raster(get_raster_path("d13C_cel_map_BRAZIL_stack.tiff"), "d13C_var", use_only_band_index=1)
+  return d13C_var_geotiff_
+
+# A collection of column names to functions that load the corresponding geotiffs.
+column_name_to_geotiff_fn = {
```
Comment (erickzul) [inline raster.py:436]: Nit: I think that this dict and the methos above that we use can be encapsulated in a class and that would make this easier to read.

**Review by erickzul — CHANGES_REQUESTED** [review]
Some nits and comments to consider making changes for.

**Pull request closed without merging**